### PR TITLE
Adding a README

### PR DESCRIPTION
**Pull request by hmemcpy**
Connects to #29 

This adds a brief explanation and some getting started snippets to the readme.,

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
...
+[Test]
+public async Task ShouldReplyWithResponseMessage()
+{
+    var handler = new MyHandler();
+    var fakeContext = new TestableMessageHandlerContext();
+
+    await handler.Handle(new RequestMessage(), fakeContext);
+
+    Assert.AreEqual(1, fakeContext.RepliedMessages.Length);
+    Assert.IsInstanceOf<ResponseMessage>(fakeContext.RepliedMessages[0].Message);
```
Comment (timbussmann) [inline README.md]: could that be replaced with `Assert.AreEqual(1, fakeContext.RepliedMessages.Containing<ResponseMessage>().Count())`?

**Comment by timbussmann** [comment]
I think a link to the corresponding docs pages would be helpful

File: README.md (hunk 1 above)
Comment (hmemcpy) [inline README.md]: @timbussmann maybe, but that relies even more on the extension methods. This one shows the real bare minimum (both tests make it obvious `RepliedMessages` is an array/list.

**Comment by hmemcpy** [comment]
Yeah, I'll add a link :+1:

File: README.md (hunk 1 above)
Comment (timbussmann) [inline README.md]: I can live with that. Just wondering how users will discover those extensions then?

**Comment by danielmarbach** [comment]
I really question the value of this. Now we have to maintain snippets in the doc site and in this readme. Can't we just link to the docs site or some sample in the readme?

// cc @Particular/docs-maintainers

**Comment by hmemcpy** [comment]
When coming to a public repo, I would love to have all the info I need to get started quickly just from the readme. While this may be a maintenance issue, I think seeing some quick code bits is very helpful.
Otherwise, why have this in a separate public repo at all?

**Comment by SimonCropp** [comment]
We have a seperate repo for deployment, versioning and issue tracking. Not for doco.

+1 to link to doco

**Comment by SimonCropp** [comment]
This will be an ongoing maintenence pain.

**Comment by hmemcpy** [comment]
Alright, I will remove the snippets and leave just the link to doco.

What about a license file? I think NSB.Testing should have the docs/snippets license. Apache, was it?

**Comment by SimonCropp** [comment]
why would it have the same license as doco? it is not doco

**Comment by hmemcpy** [comment]
I don't think this should have a restrictive license (i.e. not that of NSB). So either MIT or the one we use for doco, which is Apache.

**Comment by SimonCropp** [comment]
> I don't think this should have a restrictive license

why would i be different than all our other repos?

**Comment by hmemcpy** [comment]
Don't know, many of our public repos don't have an explicit license file btw

**Comment by andreasohlund** [comment]
Inspections failing

**Comment by hmemcpy** [comment]
Sent a separate PR (#78) that adds the NSB license.

**Pull request merged**